### DISCUSSION TITLE
Make `arbitrary_types_allowed` default to True with dataclass decorator

### DISCUSSION THREAD

**Discussion by jzwick-narmi** [pydantic/pydantic#3744] [Feature Request]
from the docs:
> pydantic.dataclasses.dataclass is a drop-in replacement for dataclasses.dataclass with validation

But needing to define a config and pass it to all my already defined dataclasses in order for them to work out of the box makes it strictly not a drop-in replacement. Plus, I don't see why this would be restricted to begin with, why would it be useful NOT to allow arbitrary class types? Seems niche to me.

If there is no interest in defaulting  to`True` here, perhaps update the documentation with this caveat.

**Comment by rusheb** [comment]
For anybody who (like me) came here trying to figure out how to set `arbitrary_types_allowed` for pydantic dataclasses, the solution/workaround is to use the [config](https://docs.pydantic.dev/latest/concepts/dataclasses/#dataclass-config) parameter:
```python
from pydantic import ConfigDict
from pydantic.dataclasses import dataclass

class MyClass:
    pass

@dataclass(config=ConfigDict(arbitrary_types_allowed=True))
class MyPydanticDataclass:
    a: MyClass
 ```
